src/Tests/AzureClientMagicTests.cs

```diff
@@ -268,6 +268,7 @@ public class MockAzureClient : IAzureClient
         internal List<string> SubmittedJobs = new List<string>();
         internal List<string> ExecutedJobs = new List<string>();
 
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
```

Comment:
Would it make sense to wrap this method in a sync method with a wait to avoid changing compiler pragmas?

Reply:
Since this is test code, I think disabling this warning is acceptable to improve readability of the code. There are several other test files in this project that already disable this particular warning; we have many interfaces with async methods, and the tests often have stubs of these that aren't actually asynchronous. (I definitely agree with you that it'd be better not to do this in product code.)